FireboltNETSDK/FireboltClient2.cs

```diff
@@ -104,11 +106,19 @@ private async Task<T> GetAccountField<T>(string fieldName, string accountName)
     public override async Task<ConnectionResponse> ConnectAsync(string? engineName, string database, CancellationToken cancellationToken)
     {
         await EstablishConnection();
-        // Connecting to system engine by default
-        var result = await GetSystemEngineUrl(_account);
-        if (result.engineUrl != null)
+        string cacheKey = $"{_env}.{_account}";
+        string? systemEngineUrl;
+        systemEngineUrlCache.TryGetValue(cacheKey, out systemEngineUrl);
+        if (systemEngineUrl == null)
+        {
+            // Connecting to system engine by default
```

Comment:
nit: Let's move this comment where i makes more sense, otherwise in here it's confusing since even without going into the if block we're still connecting to system engine by default.

Reply:
When you added this comment on 2023-07-26 it did not confuse you. 

e3f72f6f (Petro Tiurin     2023-07-26 09:41:47 +0100 257)             // Connecting to system engine by default

Now, the only thing that was changed is an indentation because of the caching. 

Here is the original code:

```
        public override async Task<bool> OpenAsync(CancellationToken cancellationToken)
        {
            if (_fireboltClient == null)
            {
                _fireboltClient = new FireboltClient(ClientId, ClientSecret, Endpoint, Env, HttpClientSingleton.GetInstance());
            }
            await Client.EstablishConnection();
            // Connecting to system engine by default
            var result = await Client.GetSystemEngineUrl(Account);
            EngineUrl = result.engineUrl;
            // Specific engine and database specified
            if (EngineName != null && Database != string.Empty)
            {
                EngineUrl = GetEngineUrlByEngineNameAndDb(EngineName, Database);
                _isSystem = false;
            }
            else if (EngineName != null)
            {
                // If no db provided - try to fetch it
                var database = GetEngineDatabase(EngineName);
                if (database == null)
                {
                    throw new FireboltException($"Engine {EngineName} is attached to a database current user can not access");
                }
                EngineUrl = GetEngineUrlByEngineNameAndDb(EngineName, database);
                _database = database;
                _isSystem = false;
            }
            OnSessionEstablished();
            return EngineUrl != null;
        }



```
Where this comment makes more sense? Please let me know and I will move it.

Reply:
> When you added this comment on 2023-07-26 it did not confuse you.

Nope, it did not.

Now, if you do a git blame on the current version you'll find out that you in fact changed this section (also indicated by the green background in github). Please re-read the code and the comment above to find a better place for it since it no longer makes sense.

Reply:
This code was just moved from Connection to FireboltClient2. The only thing that I changed now is adding the cache. It is really not a problem to remove this comment. I really do not know where is the better place for it. Please suggest if you do no agree that the comment that you wrote yourself is misplaced.

Reply:
pushed this [change](https://github.com/firebolt-db/firebolt-net-sdk/pull/89/commits/36c8cd5a658910cd99b1b95eee4ec469581e35ca) to your branch, please remember to sync.
As I said, you've added an if around it. It makes sense to move the comment out since we're either way (in and out of if) connecting to system engine (or remove it altogether). 
It does not matter who wrote the code, you touch it - you own it, as simple as that. I really don't understand how this small nitpick has become such an issue.